include/dsn/dist/distributed_lock_service.h

```diff
@@ -69,7 +69,7 @@ namespace dsn
             /*
              * initialization routine
              */
-            virtual error_code initialize() = 0;
+            virtual error_code initialize(const char* work_dir, const char* lock_root) = 0;
```

Comment:
This is way too specific to the implementation. What if we come up another lock service atop of an existing watchable storage, e.g., etcd, in that case, such parameters are not appropriate. I would suggest std:string as args to be more general.